File: medicalcare_frontend/src/app/models/patient.ts
export class Patient {
    constructor(
        public id: number,
        public code: string,
        public first_name: string,
        public last_name: string,
        public date_of_birth: Date,
        public gender: string,
        public home_address: string,
        public office_address: string,
        public phone_number: string,
        public email: string,
        public job: string,
        public emergency_contact_name: string,
        public emergency_contact_phone: string,
        public insurance_type: string,
        public insurance_policy_number: string,
        public insurance_provider: string,
        public insurance_expire: Date,
        public insurance_info: string,
        public medical_history: string
    ){

    }
}

File: Medicalcare_API/Models/Patient.cs

using System.ComponentModel.DataAnnotations;

namespace Medicalcare_API.Models{
    public class Patient{
        
        [Key]
        public int id{get;set;}

        [MaxLength(50)]
        public string? email{get;set;}

        [Required]
        [MaxLength(50)]
        public string? code{get;set;}
        

        [Required]
        [MaxLength(50)]
        public string? first_name{get;set;}

        [Required]
        [MaxLength(50)]
        public string? last_name{get;set;}

        public DateTime date_of_birth{get;set;}
        public string? gender{get;set;}
        public string? home_address{get;set;}
        public string? office_address{get;set;}
        public string? phone_number{get;set;}
        public string? job{get;set;}
        public string? emergency_contact_name{get;set;}
        public string? emergency_contact_phone{get;set;}
        public string? insurance_type{get;set;}
        public string? insurance_policy_number{get;set;}
        public string? insurance_provider{get;set;}
        public DateTime insurance_expire{get;set;}
        public string? insurance_info{get;set;}
        public string? medical_history{get;set;}
   }
}